doc/reference/vm_live_migration_internals.md

```diff
@@ -0,0 +1,104 @@
+1. Stop and delete source VM.
+1. Start the target VM using the transferred state.
+
+## Migration API
+
+Sending a `POST` request to `/1.0/instances/{name}` renames, moves an instance between pools, or migrates an instance to another server. In the push case, the returned operation metadata for migration is a background operation with progress data. For the pull case, it is a WebSocket operation with a number of secrets to be passed to the target server.
+
+## Live migration call stack
+
+Below is a general overview of the key functions of the live migration call stack:
+
+### [`lxd/lxd/instance_post.go`](https://github.com/canonical/lxd/blob/main/lxd/instance_post.go)
+
+[`instancePost`](https://github.com/canonical/lxd/blob/main/lxd/instance_post.go#L74)
+
+This function handles post requests to the `/1.0/instances` endpoint.
+
+### [`lxd/lxd/migrate_instance.go`](https://github.com/canonical/lxd/blob/main/lxd/migrate_instance.go)
+
+[`Do`](https://github.com/canonical/lxd/blob/main/lxd/migrate_instance.go#L87)
+
+This function performs the migration operation on the source VM for the given state and operation. It sets up the necessary WebSocket connections for control, state, and filesystem, and then initiates the migration process.
+
+### [`lxd/lxd/instance/drivers/driver_qemu.go`](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go)
+
+[`MigrateSend`](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go#L6436)
+
+This function controls the sending of a migration, checking for stateful support, waiting for connections, performing checks, and sending a migration offer. When performing an intra-cluster same-name migration, steps are taken to prevent corruption of volatile device configuration keys during the start and stop of the instance on both source and target.
+
+[`migrateSendLive`](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go#L6666)
+
+This function performs the live migration send process:
+
+1. Connect to the QEMU monitor: The function begins by establishing a connection to the QEMU monitor using `qmp.Connect`.
+1. Define disk names: The function defines names for the root disk (`lxd_root`), the NBD target disk (`lxd_root_nbd`), and the snapshot disk (`lxd_root_snapshot`). These will be used later to manage the root disk and its snapshot during migration.
+1. Check for shared storage: If the migration involves shared storage, the migration process can bypass the need for synchronizing the root disk. The function checks for this condition by verifying if `clusterMoveSourceName` is non-empty and the pool is remote.
+1. Non-shared storage snapshot setup: If shared storage is not used, the function proceeds to set up a temporary snapshot of the root disk.
+  - Migration capabilities such as `auto-converge`, `pause-before-switchover`, and `zero-blocks` are set to optimize the migration process.
```

Comment:
Here, the bullets are rendering as:
![image](https://github.com/user-attachments/assets/b6d0bc03-035f-4870-abe2-0288123ca08b)

Reply:
That way we can use the dynamic numbering starting each top level point with `1.`.

Reply:
It looks like indenting 4 spaces as shown below works well:

```
1.
1.
    1.
    1.
```